sw/device/lib/testing/clkmgr_testutils.c

```diff
@@ -6,27 +6,102 @@
+    const dif_clkmgr_t *clkmgr, dif_clkmgr_hintable_clock_t clock,
     bool exp_clock_enabled, uint32_t timeout_usec);
 
-extern void clkmgr_testutils_enable_clock_count_measurement(
-    dif_clkmgr_t *clkmgr, dif_clkmgr_measure_clock_t clock,
-    uint32_t lo_threshold, uint32_t hi_threshold);
+// The thresholds are encoded as
+// - max = count + variability
+// - min = count - variability
+typedef struct expected_count_info {
+  uint32_t count;
+  uint32_t variability;
+} expected_count_info_t;
 
-void clkmgr_testutils_disable_clock_count_measurements(dif_clkmgr_t *clkmgr) {
+static const expected_count_info_t kNoJitterCountInfos[] = {
+    {.count = 479, .variability = 1}, {.count = 239, .variability = 1},
+    {.count = 119, .variability = 1}, {.count = 499, .variability = 1},
+    {.count = 239, .variability = 1},
+};
+
+// If jitter is enabled the low threshold should be up to 20% lower, so
+// the variability is set to 0.1 * max_count, and count as max - 0.1 * max.
+static const expected_count_info_t kJitterCountInfos[] = {
+    {.count = 480 - 48, .variability = 48},
+    {.count = 240 - 24, .variability = 24},
+    {.count = 120 - 12, .variability = 12},
+    {.count = 500 - 50, .variability = 50},
+    {.count = 240 - 24, .variability = 24},
+};
+
+void clkmgr_testutils_enable_clock_count(const dif_clkmgr_t *clkmgr,
+                                         dif_clkmgr_measure_clock_t clock,
+                                         uint32_t lo_threshold,
+                                         uint32_t hi_threshold) {
+  LOG_INFO("Enabling clock count measurement for %s(%d) lo %0d hi %0d",
+           measure_clock_names[clock], clock, lo_threshold, hi_threshold);
+  CHECK_DIF_OK(dif_clkmgr_enable_measure_counts(clkmgr, clock, lo_threshold,
+                                                hi_threshold));
+}
+
+void clkmgr_testutils_enable_clock_counts_with_expected_thresholds(
+    const dif_clkmgr_t *clkmgr, bool jitter_enabled, bool external_clk,
+    bool low_speed) {
+  CHECK(!(external_clk && jitter_enabled));
+  for (int clk = 0; clk < ARRAYSIZE(kNoJitterCountInfos); ++clk) {
+    const expected_count_info_t *count_info;
+    if (jitter_enabled) {
+      count_info = &kJitterCountInfos[clk];
+    } else if (external_clk) {
+      if (low_speed) {
+        if (clk == kDifClkmgrMeasureClockIo ||
+	    clk == kDifClkmgrMeasureClockMain) {
+          count_info = &kNoJitterCountInfos[kDifClkmgrMeasureClockIoDiv2];
+        } else {
+	  count_info = &kNoJitterCountInfos[clk];
+	}
+      } else {
+        if (clk == kDifClkmgrMeasureClockMain) {
+          count_info = &kNoJitterCountInfos[kDifClkmgrMeasureClockIo];
+        } else {
+	  count_info = &kNoJitterCountInfos[clk];
+	}
+      }
+    } else {
+      count_info = &kNoJitterCountInfos[clk];
+    }
+    clkmgr_testutils_enable_clock_count(
+        clkmgr, (dif_clkmgr_measure_clock_t)clk,
+        count_info->count - count_info->variability,
```

Comment:
do you minus `variability` twice?
> {.count = 480 - 48, .variability = 48},

Reply:
Yes. See comment in lines 22 - 24 and 36 - 37.